app/views/courses/_walkout_button.html.erb

```diff
@@ -0,0 +1,5 @@
+<% if signed_in? && current_user.student? %>
+  <%= link_to @student, method: :delete, data: { confirm: "Вы уверены, что хотите покинуть курс?" } do %>
+    <span class="btn btn-danger">Отписаться</span>
```

Comment:
Why not store something like `active` field somewhere in db instead of moving lots of other data?
I think the best way to do this is to implement another model that will hold all user subscriptions for a particular course and will have this new field for unsubscribing.